db/migrate/20180503141759_modify_scheme10_offence_data.rb

```diff
@@ -0,0 +1,71 @@
+class ModifyScheme10OffenceData < ActiveRecord::Migration[5.0]
+  # NOTE: smoke test on build of branch means we
+  # need to cater for offence not existing --> update_all
+  def up
+    relation.
+      where(description: 'Causing a person to engage in sexual activity without consent: Sexual Offences Act 2003, s.4(4)').
+      update_all(description: 'Causing a person to engage in sexual activity without consent')
+
+    relation.
+      where('description ILIKE ?', 'Demanding payment for the inclusion of a person%').
+      update_all(description: 'Directory entries')
+
+    relation.
+      where(description: 'Failure to comply with a remedial order.').
+      update_all(description: 'Failing to comply with a remedial order')
+
+    relation.
+      where(description: 'Give to another person disclosed protected material in connection with sexual offence proceedings.').
+      update_all(description: 'Give that (protected) material or any copy of it, or otherwise reveal its contents, to any other person')
+
+    relation.
+      where(description: 'Offences arising from breach of regulation 20 (Monies in Trust) and 21 (Monies in Trust where other party to contract is acting otherwise than in the court of business).').
+      update_all(description: 'Offences arising from breach of Regulations 20 and 21')
+
+    relation.
+      where(description: 'Offences under regulations 6 (Borrowing and banking of landfill allowances) and 7 (Trading and other transfer of landfill allowances).').
+      update_all(description: 'Offences under regulations under s.6 and s.7')
+
+    relation.
+      where(description: 'Possession or supply of apparatus for use in dishonestly obtaining an electronic communication service.').
+      update_all(description: 'Possession or supply of apparatus etc for contravening s.125')
+
+    Rake::Task['data:migrate:offence_unique_code_scheme_10'].invoke
```

Comment:
🙅‍♂️ Mixing migrations with rake task IMO is **not** a good idea and will come back to bite us eventually.

This ties that rake task as a dependency of the migration, they should not be related/dependent on each other. 

💭 Consider having the rake task behaviour as a service under `<RAILS_ROOT>/db/data` so it remains completely under the data migration scope.

Reply:
its complicated